packages/destination-actions/src/destinations/delivrai-activate/updateSegment/index.ts

```diff
@@ -0,0 +1,153 @@
+      description:
+        'If true, batch requests to Delivr AI. Delivr AI accepts batches of up to 1000 events. If false, send each event individually.',
+      type: 'boolean',
+      default: true,
+      unsafe_hidden: true
+    },
+    batch_size: {
+      label: 'Batch Size',
+      description: 'Maximum number of events to include in each batch. Actual batch sizes may be lower.',
+      type: 'number',
+      default: 1000,
+      unsafe_hidden: true
+    }
+  },
+
+  perform: async (request, { payload, settings }) => {
+      // Check if client_identifier_id is valid
+      if (!settings?.client_identifier_id) {
+          return [{ status: false, message: "Client identifier is required" }];
+      }
+
+      // Generate JWT token
+      const rt_access_token_response = await generate_jwt(settings.client_identifier_id, request);
+      
+      // Check if token generation was successful
+      if (rt_access_token_response === 401) {
+          return [{ status: false, message: "Not valid customer" }];
+      }
+
+      const rt_access_token = rt_access_token_response?.data?.token;
+
+      // Process the payload with the valid token
+      return process_payload(request, [payload], rt_access_token, settings.client_identifier_id);
+  },
+  performBatch: async (request, { payload, settings }) => {
+    // Ensure client_identifier_id is provided
+    if (!settings?.client_identifier_id) {
+        return [{ status: false, message: "Client identifier is required" }];
+    }
+    try {
+        // Generate JWT token
+        const rt_access_token_response = await generate_jwt(settings.client_identifier_id, request);
+        // Check if token generation was successful
+        if (rt_access_token_response?.status === 401) {
+            return [{ status: false, message: "Not valid customer" }];
+        }
+        const rt_access_token = rt_access_token_response?.data?.token;
+        // Process the payload with the valid token
+        return process_payload(request, payload, rt_access_token, settings.client_identifier_id);
+    } catch (error:any) {
+        // Handle unexpected errors
+        return [{ status: false, message: "An error occurred", error: error?.message }];
+    }
+  }
+}
+
+// Makes a request to Delivr AI Realtime API to populate an audience
+async function process_payload(
+  request: RequestClient,
+  payload: Payload[],
+  token: string | undefined,
+  client_identifier_id: string
+) {
+  const body = gen_update_segment_payload(payload, client_identifier_id)
+  // Send request to Delivr AI only when all events in the batch include selected Ids
+  if (body.data.length > 0) {
+    return await request(`${DELIVRAI_BASE_URL}${DELIVRAI_SEGMENTATION_AUDIENCE}`, {
+      method: 'POST',
+      json: body,
+      headers: {
+        Authorization: `Bearer ${token}`
+      }
+    }).catch((err) => {
+      return [{ status: false, message: 'HTTP error: ' + err.response.message , code: err.response.statusCode}];
```

Comment:
```suggestion
      throw new APIError('HTTP error: ' + err.response.message, err.response.statusCode)
```
Use built in Segment error for better observability